### PR TITLE
Bug fixes and recursive searching for all matches

### PR DESCRIPTION
**Pull request by dxvidnrt**
This implements:

- a feature to search all fitting matches 
  recursively
- an endpoint `get_all_matches` of the server
- a bug fix, changing the listen address from 
  `127.0.0.1` to `0.0.0.0` to work inside of 
  docker networks
- a bug fix, to not empty the matches list and 
  actually return all relevant matches

### CODE REVIEW

**Review by s-heppner — CHANGES_REQUESTED** [review]
I think the biggest issue I see at the moment is the cross-dependency you introduced to the semantic_id_resolver. 
If possible at all, I would avoid calling that in as a dependency. 

If that doesn't work out, my suggestion is to create a completely new repository for the test framework, where you pull in both projects as dependency.

File: semantic_matcher/service.py [hunk 1]
```diff
@@ -38,6 +38,17 @@ def __init__(
             equivalences that this :class:`~.SemanticMatchingService` contains.
         """
         self.router = APIRouter()
+
+        self.router.add_api_route(
+            "/",
+            self.read_root,
+            methods=["GET"]
+        )
```
[the remark this answers is not in the document]
Comment (dxvidnrt) [reply semantic_matcher/service.py]: Yes, removed it

File: config.ini.default [hunk 2]
```diff
@@ -1,4 +1,8 @@
 [SERVICE]
 endpoint=http://127.0.0.1
 port=8000
-equivalence_table_file=./data/equivalence_table.json
\ No newline at end of file
+equivalence_table_file=./data/equivalence_table.json
+
+[RESOLVER]
```
[the remark this answers is not in the document]
Comment (dxvidnrt) [reply config.ini.default:6]: If I understand you correctly, we add information to the semantic_id itself.
The current structure is: "SMS_ID"/semanticID/"ID".
So we could add the corresponding resolver endpoint: "Resolver_Endpoint"/"SMS_ID"/semanticID/"ID".

[the remark this answers is not in the document]
Comment (s-heppner) [reply config.ini.default:6]: Nevermind, we do need to specify the resolver on the Semantic Matching Service level.

File: semantic_matcher/service.py [hunk 3]
```diff
@@ -48,9 +54,27 @@ def __init__(
...
+            "/clear",
+            self.remove_all_matches,
+            methods=["POST"]
+        )
         self.endpoint: str = endpoint
         self.equivalence_table: model.EquivalenceTable = equivalences
 
+    def read_root(self):
+        return {"message": "Hello, World!"}
+
```
Comment (s-heppner) [inline semantic_matcher/service.py:67]: Should be removed as well

File: requirements.txt [hunk 4]
```diff
@@ -2,3 +2,4 @@ fastapi>=0.95.0
 pydantic>=1.10
 uvicorn>=0.21.1
 requests>=2.31.0
+git+https://github.com/dxvidnrt/semantic_id_resolver
```
[the remark this answers is not in the document]
Comment (dxvidnrt) [reply requirements.txt]: Dependency removed

File: semantic_matcher/service.py [hunk 5]
```diff
@@ -109,7 +143,23 @@ def _get_matcher_from_semantic_id(self, semantic_id: str) -> str:
...
+        if response.status_code == 200:
+            # Parse the JSON response and construct SMSResponse object
+            response_json = response.json()
+            sms_response = resolver_service.SMSResponse(
+                semantic_matching_service_endpoint=response_json['semantic_matching_service_endpoint'],
+                meta_information=response_json['meta_information']
+            )
+            return sms_response.semantic_matching_service_endpoint
+
+        return None
```
[the remark this answers is not in the document]
Comment (dxvidnrt) [reply semantic_matcher/service.py]: Dependency resolved, implemented as dict

File: semantic_matcher/service.py (hunk 3 above)
Comment (dxvidnrt) [reply semantic_matcher/service.py]: Removed

File: requirements.txt [hunk 6]
```diff
@@ -1,4 +1,4 @@
 fastapi>=0.95.0
 pydantic>=1.10
 uvicorn>=0.21.1
-requests>=2.31.0
+requests>=2.31.0
```
Comment (s-heppner) [inline requirements.txt:4]: Can you reintroduce the empty newline at the end of the file?

**Review by s-heppner — CHANGES_REQUESTED** [review]
Except for this minor thing, it looks good to me.

**Comment by s-heppner** [comment]
I took the liberty to improve your PR title and message, so that we can use it as commit message, when we squash and merge.

**Review by s-heppner — APPROVED** [review]
LGTM now

**Pull request merged**